app/src/main/java/org/mozilla/fenix/onboarding/DefaultBrowserNotificationWorker.kt

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Context.NOTIFICATION_SERVICE
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationCompat.BADGE_ICON_NONE
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.content.ContextCompat
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import mozilla.components.support.base.ids.SharedIdsHelper
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.utils.Settings
+
+class DefaultBrowserNotificationWorker(
+    context: Context,
+    workerParameters: WorkerParameters
+) : Worker(context, workerParameters) {
+
+    override fun doWork(): Result {
+        ensureChannelExists()
+        NotificationManagerCompat.from(applicationContext)
+            .notify(NOTIFICATION_TAG, NOTIFICATION_ID, buildNotification())
+        return Result.success()
+    }
+
+    /**
+     * Build the notification with site controls to be displayed while the web app is active.
+     */
+    private fun buildNotification(): Notification {
+        val channelId = ensureChannelExists()
+        val intent = Intent(applicationContext, HomeActivity::class.java)
+        intent.putExtra(INTENT_DEFAULT_BROWSER_NOTIFICATION, true)
+
+        val pendingIntent = PendingIntent.getActivity(
+            applicationContext,
+            SharedIdsHelper.getNextIdForTag(applicationContext, NOTIFICATION_PENDING_INTENT_TAG),
+            intent,
+            0
+        )
+
+        with(applicationContext) {
+            val appName = getString(R.string.app_name)
+            return NotificationCompat.Builder(this, channelId)
+                .setSmallIcon(R.drawable.ic_status_logo)
+                .setContentTitle(
+                    applicationContext.getString(R.string.notification_default_browser_text, appName))
+                .setBadgeIconType(BADGE_ICON_NONE)
+                .setColor(ContextCompat.getColor(this, R.color.primary_text_light_theme))
+                .setPriority(NotificationCompat.PRIORITY_LOW)
+                .setShowWhen(false)
+                .setContentIntent(pendingIntent)
+                .build()
+        }
+    }
+
+    /**
+     * Make sure a notification channel for the powered by notifications exists.
```

Comment:
Docs from a different place?

Reply:
Seems like this class is coming from `PoweredByNotification`. Is there anything around here that would be good to abstract out for re-use?

Reply:
Yes, this might be a good abstraction for the support library.   Currently there's a lot of `ensureChannelExists` function calls in both Fenix and Android Components.  I'll created an issue to track this work.

Reply:
https://github.com/mozilla-mobile/android-components/issues/10520